x-pack/plugins/transform/public/app/sections/transform_management/components/action_edit/use_edit_action.tsx

```diff
@@ -5,25 +5,70 @@
  * 2.0.
  */
 
-import React, { useContext, useMemo, useState } from 'react';
+import React, { useCallback, useContext, useMemo, useState } from 'react';
 
-import { TransformConfigUnion } from '../../../../../../common/types/transform';
+import { i18n } from '@kbn/i18n';
 
 import { TransformListAction, TransformListRow } from '../../../../common';
 import { AuthorizationContext } from '../../../../lib/authorization';
 
 import { editActionNameText, EditActionName } from './edit_action_name';
+import { useSearchItems } from '../../../../hooks/use_search_items';
+import { useAppDependencies, useToastNotifications } from '../../../../app_dependencies';
+import { TransformConfigUnion } from '../../../../../../common/types/transform';
 
 export const useEditAction = (forceDisable: boolean, transformNodes: number) => {
   const { canCreateTransform } = useContext(AuthorizationContext).capabilities;
 
   const [config, setConfig] = useState<TransformConfigUnion>();
   const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+  const [indexPatternId, setIndexPatternId] = useState<string | undefined>();
+
   const closeFlyout = () => setIsFlyoutVisible(false);
-  const showFlyout = (newConfig: TransformConfigUnion) => {
-    setConfig(newConfig);
-    setIsFlyoutVisible(true);
-  };
+
+  const { getIndexPatternIdByTitle, loadIndexPatterns } = useSearchItems(undefined);
+  const toastNotifications = useToastNotifications();
+  const appDeps = useAppDependencies();
+  const savedObjectsClient = appDeps.savedObjects.client;
+  const indexPatterns = appDeps.data.indexPatterns;
+
+  const clickHandler = useCallback(
+    async (item: TransformListRow) => {
+      try {
+        await loadIndexPatterns(savedObjectsClient, indexPatterns);
+        const indexPatternTitle = Array.isArray(item.config.source.index)
+          ? item.config.source.index.join(',')
+          : item.config.source.index;
+        const currentIndexPatternId = getIndexPatternIdByTitle(indexPatternTitle);
+
+        if (currentIndexPatternId === undefined) {
+          toastNotifications.addWarning(
+            i18n.translate('xpack.transform.edit.noIndexPatternErrorPromptText', {
+              defaultMessage:
+                'Unable to get index pattern the transform {transformId}. No index pattern exists for {indexPattern}.',
+              values: { indexPattern: indexPatternTitle, transformId: item.id },
+            })
+          );
+        }
+        setIndexPatternId(currentIndexPatternId);
+        setConfig(item.config);
+        setIsFlyoutVisible(true);
+      } catch (e) {
+        toastNotifications.addDanger(
+          i18n.translate('xpack.transform.clone.errorPromptText', {
+            defaultMessage: 'An error occurred checking if source index pattern exists',
+          })
+        );
```

Comment:
you should pass the actual error object as well

Reply:
`displayErrorToast` from `toastNotificationServiceProvider` could be used as it'll extract the error for you.

Reply:
`toastNotificationServiceProvider` is not currently in Transform, so I used `addError` for now. I can bring the service over if we need to use it more in other places https://github.com/elastic/kibana/pull/108804/commits/6892992839d51dfdf566ce9ba5bdfae4c95c5329